src/component/feature/progress.jsx

```diff
@@ -23,16 +24,78 @@ class Progress extends Component {
     }
 
     componentWillReceiveProps ({ percentage }) {
-        this.setState({ percentage });
+        if (this.state.percentage !== percentage) {
+            const nextState = { percentage };
+            if (this.props.animatePercentageText) {
+                this.animateTo(percentage, this.getTarget(percentage));
+            } else {
+                nextState.percentageText = percentage;
+            }
+            this.setState(nextState);
+        }
+    }
+
+    getTarget (target) {
```

Comment:
The speed of the "animated wheel" does not match the speed of the animated text. 

![image](https://cloud.githubusercontent.com/assets/158948/21287316/eac43aec-c468-11e6-910e-3ae26cd3d326.png)